### PR TITLE
Add scheme to destination region endpoint

### LINKED ISSUE #1564: EC2 copy EBS snapshot missing protocol.
When I try to use the `copySnapshot()` method with the EC2 client, I get a null pointer exception with the message that the "protocol must not be null"

## Expected Behavior
The EBS snapshot should be copied from one region to another.

## Current Behavior
Currently,  I get the following error:
```
Exception in thread "main" java.lang.NullPointerException: protocol must not be null.
	at software.amazon.awssdk.utils.Validate.paramNotNull(Validate.java:117)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest.standardizeProtocol(DefaultSdkHttpFullRequest.java:63)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest.<init>(DefaultSdkHttpFullRequest.java:52)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest.<init>(DefaultSdkHttpFullRequest.java:41)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest$Builder.build(DefaultSdkHttpFullRequest.java:331)
	at software.amazon.awssdk.http.DefaultSdkHttpFullRequest$Builder.build(DefaultSdkHttpFullRequest.java:170)
	at software.amazon.awssdk.services.ec2.transform.internal.GeneratePreSignUrlInterceptor.modifyHttpRequest(GeneratePreSignUrlInterceptor.java:104)
	at software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain.modifyHttpRequestAndHttpContent(ExecutionInterceptorChain.java:99)
	at software.amazon.awssdk.core.client.handler.BaseClientHandler.runModifyHttpRequestAndHttpContentInterceptors(BaseClientHandler.java:123)
	at software.amazon.awssdk.core.client.handler.BaseClientHandler.finalizeSdkHttpFullRequest(BaseClientHandler.java:68)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:106)
	at software.amazon.awssdk.core.client.handler.BaseSyncClientHandler.execute(BaseSyncClientHandler.java:73)
	at software.amazon.awssdk.core.client.handler.SdkSyncClientHandler.execute(SdkSyncClientHandler.java:44)
	at software.amazon.awssdk.awscore.client.handler.AwsSyncClientHandler.execute(AwsSyncClientHandler.java:55)
	at software.amazon.awssdk.services.ec2.DefaultEc2Client.copySnapshot(DefaultEc2Client.java:3020)
	at software.amazon.awssdk.services.ec2.Ec2Client.copySnapshot(Ec2Client.java:3900)
	at edu.byu.artifactoryLambda.Handler.backupEbs(Handler.kt:101)
	at edu.byu.artifactoryLambda.Handler.handleRequest(Handler.kt:46)
	at edu.byu.artifactoryLambda.HandlerKt.main(Handler.kt:28)
	at edu.byu.artifactoryLambda.HandlerKt.main(Handler.kt)

Process finished with exit code 130 (interrupted by signal 2: SIGINT)
```

## Possible Solution
Include the protocol when the URL is getting signed

## Steps to Reproduce (for bugs)
```kotlin
        val latestSnapshot = ec2Client.describeSnapshots { request ->
            request.filters(Filter.builder().name("volume-id").values(currentVolumeId).build())
        }.snapshots()[0]

        ec2Client.copySnapshot { request ->
            request.description("Artifactory backup ${latestSnapshot.startTime()}")
                .sourceRegion("us-west-2")
                .sourceSnapshotId(latestSnapshot.snapshotId())
                .destinationRegion("us-east-1")
        }
```

## Context
I need to make a backup of an EBS snapshot from one region to another so I can have data recovery if needed.

## Your Environment

* AWS Java SDK version used: 2.10.41
* JDK version used: Corretto-11.0.5.10.1
* Operating System and version: Ubuntu 19.10

### PR DESCRIPTION
**Pull request by dagnir**
## Description
Fixes #1564

## Motivation and Context
Bug fix.

## Testing
New unit test, integ tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java [hunk 1]
```diff
@@ -152,6 +152,10 @@ private URI createEndpoint(String regionName, String serviceName) {
                                     .build();
         }
 
-        return Ec2Client.serviceMetadata().endpointFor(region);
+        URI endpoint = Ec2Client.serviceMetadata().endpointFor(region);
+        if (endpoint.getScheme() == null) {
+            return URI.create("https://" + endpoint);
```
Comment (spfink) [inline services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java]: Can we pull the client configured protocol in an interceptor?

Comment (dagnir) [reply services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java]: I considered this, but given that this URL is used by EC2 and not consumed by the customer/client, it doesn't really make sense to be anything other than HTTPS. It's also the same behavior as V1 https://github.com/aws/aws-sdk-java/blob/7b1e5b87b0bf03456df9e77716b14731adf9a7a7/aws-java-sdk-ec2/src/main/java/com/amazonaws/services/ec2/model/transform/GeneratePreSignUrlRequestHandler.java#L145-L147

Comment (zoewangg) [reply services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java]: +1 on pulling client configured protocol. Customers might use this against a mock server (maybe for testing purpose) which only supports HTTP. I think we should honor the customer configuration.

Comment (dagnir) [reply services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java]: I'm not sure I understand the use case. The presigned URL is explicitly used by the EC2 service to presumably pull the snapshot for cross region copying.

Comment (zoewangg) [reply services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java]: I meant there might be some mockEc2 server library available for customers to test their applications. The mockEc2 server might have a similar copySnapshot function to mimic the real service's behavior. This use case might be rare or doesn't exist at all, but I still think it'd be better to honor the customer configuration.

Comment (dagnir) [reply services/ec2/src/main/java/software/amazon/awssdk/services/ec2/transform/internal/GeneratePreSignUrlInterceptor.java]: I disagree that it's important in this case; we've had the same behavior in v1 for a long time without issue, the same with RDS. Technically the protocol configured on the client specifies the protocol it speaks to service we're calling the API on anyway, I don't think it necessarily governs things like the presigned URL(s) it generates.

IMO it's also a potential dangerous if the customer happens to being using `HTTP` and now their EBS snapshots are also being transferred in plaintext.

**Pull request merged**